micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jpa/HibernateMetrics.java

```diff
@@ -177,16 +177,29 @@ public void bindTo(MeterRegistry registry) {
         counter(registry, "hibernate.statements", "The number of prepared statements that were released",
             Statistics::getCloseStatementCount, "status", "closed");
 
-        // Second Level Caching
-        Arrays.stream(statistics.getSecondLevelCacheRegionNames())
-                .forEach(regionName ->  {
+        // Second Level Caching:
+        long numRegions = Arrays.stream(statistics.getSecondLevelCacheRegionNames())
+                .filter(regionName -> hasDomainDataRegionStatistics(regionName))
+                .map(regionName -> {
```

Comment:
I believe what you are looking for is `peek`. It is like `forEach` but it returns a stream consisting of the elements of the original stream that you can count. Also, since `peek` uses a `Consumer` (instead of a `Function`), you don't need to return anything. Also, map should not have any side-effects.

Reply:
I have another suggestion around this: to me a lambda is rather glue-code, I think would move this block into a separate method.

Reply:
I can try a few things.. this is the result of a few iterations trying to figure out how to handle the fallback case.

Reply:
I agree that peek is better. Should have thought of it. +1 for review. ;)
I would like the two blocks near each other (register these things for a region name if there are data region statistics for it, or register these other things if there were no data regions discovered)... so we could move both to a method...

Reply:
I think having a fallback is a great idea. I also think that what you did is the simplest thing that you can do on Java 8 to handle the fallback. I wanted to suggest to do a `.findAny()` and some `Optional` magic but then I realized that Java 8 does not have those magic words. :)